### PR TITLE
[RM][1105] Add new modal for adding questions

### LINKED ISSUE #1105: round manager > additional questions: add a custom question
### Background 
**Problem:** We received feedback from round operators that they wanted to have the ability to add custom questions on their round applications . Currently, all questions/answers on a  are by default text fields

**Solution:** We are giving round operators additional customizability on a round application by: 
-  enabling different question types (i.e. short/long answer, dropdown, checkboxes, etc.)
- determine if a question is required / optional 
- determine if a question is encrypted / not encrypted
- determine if the answer to a question will be visible on a project's Grant Explorer page 

**Goal:** Give round operators customization and control over their grant round applications. 

--------------
### User Story & Acceptance Criteria 

**User Story**
As a round operator, 
I want to be able to add a custom question 
so I can customize the grant round application to my program's needs 

**Acceptance Criteria**
GIVEN that I am a **_round operator_** 
WHEN I look at the Additional Questions section 
THEN I see a CTA to "Add Question" 

GIVEN that I click on "Add Question" 
WHEN I look at the page
THEN I see a modal prompting me to select a question type (short answer, paragraph answer, multiple choice, checkboxes, dropdown) 

GIVEN that I select a question type 
WHEN I look at the modal 
THEN I see more options to add Question Title 
AND options to toggle on Required (default is optional), Encrypted (default is not encrypted), and Shown in Explorer (default is Hidden from Explorer) 

GIVEN that I click "Add" 
WHEN I look at the Additional Questions section 
THEN I see the question I just added with the properties that I chose 

**UI Checklist**
Please see Figma for details. 

--------------
### Product, Design, & Engineering Notes 

**Product**
* Depending on the question type, there will be some additional steps (i.e. if the user selects "Multiple Choice, then we would have to add options for answers) and it would look slightly different after saved. 

**Design**
* Loom: https://www.loom.com/share/fcbd167f80664375ba78901c18032975
* Figma: https://www.figma.com/file/U4uRlyJJn92QiNUuyRxSvX/%5BWIP%5D-Grants-Application-%2F-Edit-Flow?node-id=65%3A2108&t=VrorsgN0AdIRBv3Y-0

**Engineering**
*

### PR DESCRIPTION
**Pull request by codenamejason**
##### Description

Adds a new modal feature for adding new questions of several types.

##### Refers/Fixes

#1105 

##### Testing
`AddQuestinModal.test.tsx`
`RoundApplicationForm.test.tsx`

### CODE REVIEW

**Comment by michellema1208** [comment]
@KurtMerbeth @codenamejason - I'm testing currently via https://discord.com/channels/562828676480237578/1075792938711400528/1075820112059301938 and noticing some UI bugs. I'm going to split up these comments based on the theme of the bugs. 

### **DEFAULT FORM**
- [ ] The icons and question types on the default form are not aligned and should be left aligned, per designs 
Screen Shot 2023-02-16 at 12 14 21 PM 

- [ ] The CTA should say "+ Add question" per designs, not "+ Add a Question" (current state) 

### **HIDDEN FROM EXPLORER / SHOWN IN EXPLORER**
- [ ] The copy on the toggle should say "Hidden from Explorer" when toggled _**off**_, and it should say "Shown in Explorer" when toggled _**on**_
- [ ] The "Shown in Explorer"/"Hidden from Explorer" logic is flipped -- when I toggle on "Shown in Explorer" and save the question, the form says "Hidden from Explorer

### **EDITING QUESTIONS** 
- [ ] When I edit a question, it saves as an entirely new question instead of updating the original question 

### **ADD QUESTION MODAL**
- [ ] Remove additional padding at the top of the Add Question modal (please see designs or refer to @willsputra for assistance on pixels/sizing) 

- [ ] Make sure the font and size of "Add Question" heading matches designs (please refer to @willsputra for design specs) 

- [ ] Remove the additional divider between "Question Type" label and dropdown 
Screen Shot 2023-02-16 at 12 23 11 PM 

- [ ] Make sure the font and size for "Question Type" matches designs and also sits on top of the dropdown (please refer to @willsputra for design specs) 

- [ ] When I select a Question Type option, I expect the selection to highlight the entire row not just where the text is (see designs) 
Screen Shot 2023-02-16 at 12 26 09 PM

vs. Designs: 
Screen Shot 2023-02-16 at 12 26 58 PM

- [ ] For the "Address" type, let's call it "Wallet Address", because otherwise it can be confused with home address which we don't want 
Question for @willsputra : what is the best icon for us to use for the "Wallet Address" type? 

- [ ] Make the dropdown longer so that we don't make it bulky for longer Question Types (i.e. Short Answer, Multiple Choice, etc. all don't fit in one line in the dropdown) 
Screen Shot 2023-02-16 at 12 29 00 PM

### **MULTIPLE CHOICE** 

- [ ] For multiple choice questions, there should by default be two options already set up (currently there is only one) 

Designs for reference: 
Screen Shot 2023-02-16 at 12 36 23 PM

### **APPLIES TO MULTIPLE CHOICE, DROPDOWN, & CHECKBOX QUESTIONS**

- [ ] When you have any questions with Options, there should be a vertical line that indicates that Options are nested within the question 

Currently: 
Screen Shot 2023-02-16 at 12 44 36 PM


vs. Design
Screen Shot 2023-02-16 at 12 43 36 PM
s: 

- [ ] When you save any questions that have Options (i.e. multiple choice, dropdown, checkbox), you should display the options on the form after you save

Currently: 
Screen Shot 2023-02-16 at 12 40 49 PM

vs. Designs: 
Screen Shot 2023-02-16 at 12 41 04 PM

**Comment by KurtMerbeth** [comment]
@willsputra 
> for (multiple choice, checkboxes, dropdown), could we hide the 'X' button if there's only one option left? since it doesn't do anything.

let's have an extra ticket for it. We should discuss/think about all the possible cases before.

File: packages/round-manager/src/features/common/__tests__/AddQuestionModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        onSave={() => {
+          /**/
+        }}
+        question={editQuestion}
+      />
+    );
+    expect(await screen.queryByTestId("add-question-modal")).toBeInTheDocument();
+  });
+  it("shows two default options when a multi-select type is selected??", async () => {
+    // todo:
```
Comment (boudra) [inline packages/round-manager/src/features/common/__tests__/AddQuestionModal.test.tsx:86]: are these tests still to be done or?

Comment (codenamejason) [reply packages/round-manager/src/features/common/__tests__/AddQuestionModal.test.tsx:86]: @boudra yes we still need to finish the tests.

**Review by michellema1208 — APPROVED** [review]

File: packages/round-manager/src/features/common/AddQuestionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        </Listbox>
+      </div>
+    )
+  }
+
+  const checkForErrors = (): boolean => {
+    setInputError([]);
+    const errors = [];
+
+    console.log("JER", questionOptions);
```
Comment (boudra) [inline packages/round-manager/src/features/common/AddQuestionModal.tsx:271]: Console log
```suggestion
```

**Review by boudra — APPROVED** [review]

**Pull request merged**